packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -66,14 +66,26 @@ class AndroidStudio implements Comparable<AndroidStudio> {
         pathsSelectorValue = jvmProperties['idea.paths.selector'] as String;
       }
     }
+
+    final int major = version?.major;
+    final int minor = version?.minor;
     final String presetPluginsPath = pathsSelectorValue == null
       ? null
-      : globals.fs.path.join(
-        globals.fsUtils.homeDirPath,
-        'Library',
-        'Application Support',
-        pathsSelectorValue,
-      );
+      : double.parse('$major.$minor') >= 4.1 
```

Comment:
The nested ternary expression is a big hard to read. I would break this up into an if (and use the separate major/minor checks like below):

String presetPluginsPath;
if (pathsSelectorValue != null) {
  if (major >= 4 && minor >= 1) {
    ...
  } else {
    ...
  }
}

Reply:
Got it!